simple/stats/embeddings.py

```diff
@@ -0,0 +1,62 @@
+# Copyright 2023 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+
+import pandas as pd
+from sentence_transformers import SentenceTransformer
+from stats.data import StatVar
+from util.filehandler import FileHandler
+from util.filehandler import LocalFileHandler
+
+_MODEL_NAME = "all-MiniLM-L6-v2"
+_DCID_COL = "dcid"
+_SENTENCE_COL = "sentence"
+
+
+def build(svs: list[StatVar],
+          embeddings_fh: FileHandler | None = None) -> pd.DataFrame:
+  """Builds embeddings based on the name, description and NL sentences of the specified SVs.
+
+    If the embeddings FileHandler is specified, it writes the embeddings to that file.
+    """
+  texts: list[str] = []
+  dcids: list[str] = []
+
+  def _maybe_add(text: str, dcid: str):
+    if text and dcid:
+      texts.append(text)
+      dcids.append(dcid)
+
+  for sv in svs:
+    _maybe_add(sv.name, sv.id)
+    _maybe_add(sv.description, sv.id)
+    for sentence in sv.nl_sentences:
+      _maybe_add(sentence, sv.id)
+
+  assert len(texts) == len(dcids)
+
+  logging.info("Encoding %s sentences", len(texts))
```

Comment:
Should we try to use / refactor-and-reuse from https://github.com/datacommonsorg/website/tree/master/tools/nl/embeddings ?

Reply:
Do we have a common repo?

The common code will basically be this method: https://github.com/datacommonsorg/website/blob/master/tools/nl/embeddings/build_embeddings.py#L85-L92

Reply:
I think more than that?   We could use validate embeddings, use the fine-tuned model if available, rely on the same naming with model name (so model name is clear), timestamp.

What you have is fine as a quick-one-off prototype though.

Reply:
I see.

Yes, reusing makes sense based on what you mentioned. We probably need a discussion on how similar or different main and custom DC should be and then extract the common parts.

Perhaps we can discuss once Jehangir is back and consider this a POC until then?

Reply:
We can chat tomm.

Reply:
For this PR ... does this code have access to our GCS paths and https://github.com/datacommonsorg/website/blob/5970947053bb81672a26fe286642d696b35c362c/deploy/nl/embeddings.yaml#L3 ?

If not, I think what you have is fine.

Reply:
It depends on the user running it. The current focus is custom DC and they won't have access to it. But when we run it in the future, we will. So the things you talked about above are valid and we need to have modes where we use fine tuned models, do validations, etc. over time.

Let's chat more tomorrow.

Reply:
hmm, custom DC enabling NL might need that access to get the base embeddings